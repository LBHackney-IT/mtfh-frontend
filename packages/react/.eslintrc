{
  "extends": "@lbh-mtfh/eslint-config",
  "parserOptions": {
    "project": "tsconfig.json"
  },
  "rules": {
    "prefer-arrow-callback": "off",
    "import/no-extraneous-dependencies": [
      "error",
      {
        "devDependencies": ["**/*/rollup.config.js"]
      }
    ],
    "jsx-a11y/anchor-is-valid": [
      "error",
      {
        "components": ["Link"],
        "aspects": ["invalidHref"]
      }
    ]
  }
}
