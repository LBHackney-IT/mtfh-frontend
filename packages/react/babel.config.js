const babel = require("../../babel.config")

module.exports = babel
