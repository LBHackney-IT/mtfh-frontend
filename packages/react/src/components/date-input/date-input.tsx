import React, { ComponentPropsWithoutRef, forwardRef } from "react";
import cn from "classnames";

import { Label } from "../label";
import { NumberInput, NumberInputProps } from "../number-input";
import styles from "./styles.module.scss";

export interface DateInputProps extends ComponentPropsWithoutRef<"div"> {
  id?: string;
  error?: string;
  required?: boolean;
  dayProps?: NumberInputProps;
  monthProps?: NumberInputProps;
  yearProps?: NumberInputProps;
  dayLabel?: string;
  monthLabel?: string;
  yearLabel?: string;
}

export const DateInput = forwardRef<HTMLDivElement, DateInputProps>(function DateInput(
  {
    id = "date-input",
    dayProps,
    monthProps,
    yearProps,
    dayLabel = "Day",
    monthLabel = "Month",
    yearLabel = "Year",
    error,
    required,
    className,
    ...props
  },
  ref,
) {
  return (
    <div
      ref={ref}
      className={cn(styles.govukDateInput, styles.lbhDateInput, className)}
      {...props}
    >
      <div className={styles.govukDateInputItem}>
        <Label htmlFor={`${id}-day`}>{dayLabel}</Label>
        <NumberInput
          id={`${id}-day`}
          className={cn(styles.govukDateInputInput, styles.govukInputWidth2)}
          name="day"
          required={required}
          maxLength={2}
          min={1}
          max={31}
          padStart={2}
          {...dayProps}
        />
      </div>
      <div className={styles.govukDateInputItem}>
        <Label htmlFor={`${id}-month`}>{monthLabel}</Label>
        <NumberInput
          id={`${id}-month`}
          className={cn(styles.govukDateInputInput, styles.govukInputWidth2)}
          name="month"
          required={required}
          maxLength={2}
          min={1}
          max={12}
          padStart={2}
          {...monthProps}
        />
      </div>
      <div className={styles.govukDateInputItem}>
        <Label htmlFor={`${id}-year`}>{yearLabel}</Label>
        <NumberInput
          id={`${id}-year`}
          className={cn(styles.govukDateInputInput, styles.govukInputWidth4)}
          name="year"
          required={required}
          maxLength={4}
          padStart={4}
          {...yearProps}
        />
      </div>
    </div>
  );
});
