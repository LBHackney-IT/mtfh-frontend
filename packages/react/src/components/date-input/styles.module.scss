@import "~lbh-frontend/lbh/base";
@import "~lbh-frontend/lbh/components/lbh-date-input/date-input";

// .mtfh-date-input {
//   display: grid;
//   grid-gap: 1em;
//   grid-template-columns: repeat(3, minmax(0px, 1fr));

//   .lbh-form-group & {
//     margin-top: 0;
//   }

//   .lbh-error-message + & {
//     margin-top: 0.375rem;
//   }
// }
