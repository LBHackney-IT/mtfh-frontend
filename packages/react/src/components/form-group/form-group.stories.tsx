import React from "react";

import { Input } from "../input";
import { FormGroup } from "./form-group";

export default {
  title: "Form Group",
  component: FormGroup,
};

export const Base = () => {
  return (
    <FormGroup id="name" label="First Name">
      <Input />
    </FormGroup>
  );
};

export const FormGroupWithHint = () => {
  return (
    <FormGroup
      id="name"
      label="First Name"
      hint="You can find this name on your passport"
    >
      <Input />
    </FormGroup>
  );
};

export const FormGroupWithRequired = () => {
  return (
    <FormGroup id="name" label="First Name" required>
      <Input />
    </FormGroup>
  );
};

export const FormGroupWithError = () => {
  return (
    <FormGroup
      id="name"
      label="First Name"
      error="First Name is required"
      required
    >
      <Input />
    </FormGroup>
  );
};
