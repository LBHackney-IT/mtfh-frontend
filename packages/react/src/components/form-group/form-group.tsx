import React, {
  Children,
  ComponentPropsWithoutRef,
  ReactElement,
  cloneElement,
  forwardRef,
  useMemo,
} from "react";
import classNames from "classnames";

import { widthOverrides } from "../../utils";
import { ErrorMessage } from "../error-message";
import { Hint } from "../hint";
import { Label } from "../label";
import styles from "./styles.module.scss";

export interface FormGroupProps extends ComponentPropsWithoutRef<"div"> {
  id: string;
  label?: string;
  name?: string;
  hint?: string;
  error?: string | false;
  required?: boolean;
  children: ReactElement;
  override?: number;
}

export const FormGroup = forwardRef<HTMLDivElement, FormGroupProps>(
  function FormGroup(
    {
      id,
      name,
      label,
      hint,
      error,
      required,
      children,
      className,
      override,
      ...props
    },
    ref
  ) {
    const formGroupClasses = classNames(
      styles.govukFormGroup,
      {
        [styles.govukFormGroupError]: !!error,
      },
      styles.lbhFormGroup,
      widthOverrides(override),
      className
    );

    const describedBy = useMemo(() => {
      const ids: string[] = [];
      if (hint) {
        ids.push(`${id}-hint`);
      }
      if (error) {
        ids.push(`${id}-error`);
      }
      return ids.join(" ");
    }, [id, hint, error]);

    return (
      <div ref={ref} id={id} className={formGroupClasses} {...props}>
        {!!label && (
          <Label htmlFor={`${id}-field`}>
            {label}
            {required ? <sup aria-hidden="true">*</sup> : ""}
          </Label>
        )}
        {!!hint && <Hint id={`${id}-hint`}>{hint}</Hint>}
        {!!error && <ErrorMessage id={`${id}-error`}>{error}</ErrorMessage>}
        {!!children &&
          Children.only(
            cloneElement(children, {
              id: `${id}-field`,
              name,
              required,
              error: !!error,
              "aria-describedby": describedBy || undefined,
              ...children.props,
            })
          )}
      </div>
    );
  }
);
