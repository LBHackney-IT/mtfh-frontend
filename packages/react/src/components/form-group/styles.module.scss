@import "~lbh-frontend/lbh/base";
@import "~govuk-frontend/govuk/objects/form-group";
@import "~lbh-frontend/lbh/objects/form-group";

.lbh-form-group .govuk-label sup {
  color: lbh-colour("lbh-error");
}
