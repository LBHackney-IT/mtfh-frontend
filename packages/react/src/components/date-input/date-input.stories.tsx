import React from "react"

import { FormGroup } from "../form-group"
import { DateInput } from "./date-input"

export default {
  title: "Date Input",
  component: DateInput,
}

export const Base = () => {
  return <DateInput id="date-input" />
}

export const DateInputWithError = () => {
  return (
    <FormGroup
      as="fieldset"
      id="date-input"
      label="Date of Birth"
      error="Date cannot be in the future"
      required
    >
      <DateInput
        dayProps={{ error: true }}
        monthProps={{ error: true }}
        yearProps={{ error: true }}
      />
    </FormGroup>
  )
}

export const DateInputWithSingleError = () => {
  return (
    <FormGroup
      id="date-input"
      label="Date of Birth"
      error="Date cannot be in the future"
    >
      <DateInput yearProps={{ error: true }} />
    </FormGroup>
  )
}
