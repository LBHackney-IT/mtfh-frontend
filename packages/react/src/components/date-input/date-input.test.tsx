import React from "react";
import { render } from "@testing-library/react";
import { testA11y } from "@lbh-mtfh/test-utils";

import { FormGroup } from "../form-group";
import { DateInput } from "./date-input";

test("it passes a11y", async () => {
  const { container } = render(
    <FormGroup id="test" name="test" label="DateInput Test">
      <DateInput id="date-input" />
    </FormGroup>
  );
  expect(container).toMatchSnapshot();
  await testA11y(container);
});

test("it renders correctly", () => {
  const { container } = render(<DateInput id="date-input" />);
  expect(container).toMatchSnapshot();
});
