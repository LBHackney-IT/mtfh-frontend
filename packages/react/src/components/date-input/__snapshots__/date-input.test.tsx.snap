// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it passes a11y 1`] = `
<div>
  <div
    class="govukFormGroup lbhFormGroup"
    id="test"
  >
    <label
      class="govukLabel lbhLabel"
      for="test-field"
    >
      DateInput Test
      
    </label>
    <div
      class="govukDateInput lbhDateInput"
      name="test"
    >
      <div
        class="govukDateInputItem"
      >
        <label
          class="govukLabel lbhLabel"
          for="date-input-day"
        >
          Day
        </label>
        <input
          aria-valuemax="31"
          aria-valuemin="1"
          class="govukInput lbhInput govukDateInputInput govukInputWidth2"
          id="date-input-day"
          name="day"
          role="spinbutton"
          type="number"
          value=""
        />
      </div>
      <div
        class="govukDateInputItem"
      >
        <label
          class="govukLabel lbhLabel"
          for="date-input-month"
        >
          Month
        </label>
        <input
          aria-valuemax="12"
          aria-valuemin="1"
          class="govukInput lbhInput govukDateInputInput govukInputWidth2"
          id="date-input-month"
          name="month"
          role="spinbutton"
          type="number"
          value=""
        />
      </div>
      <div
        class="govukDateInputItem"
      >
        <label
          class="govukLabel lbhLabel"
          for="date-input-year"
        >
          Year
        </label>
        <input
          class="govukInput lbhInput govukDateInputInput govukInputWidth4"
          id="date-input-year"
          name="year"
          role="spinbutton"
          type="number"
          value=""
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`it renders correctly 1`] = `
<div>
  <div
    class="govukDateInput lbhDateInput"
  >
    <div
      class="govukDateInputItem"
    >
      <label
        class="govukLabel lbhLabel"
        for="date-input-day"
      >
        Day
      </label>
      <input
        aria-valuemax="31"
        aria-valuemin="1"
        class="govukInput lbhInput govukDateInputInput govukInputWidth2"
        id="date-input-day"
        name="day"
        role="spinbutton"
        type="number"
        value=""
      />
    </div>
    <div
      class="govukDateInputItem"
    >
      <label
        class="govukLabel lbhLabel"
        for="date-input-month"
      >
        Month
      </label>
      <input
        aria-valuemax="12"
        aria-valuemin="1"
        class="govukInput lbhInput govukDateInputInput govukInputWidth2"
        id="date-input-month"
        name="month"
        role="spinbutton"
        type="number"
        value=""
      />
    </div>
    <div
      class="govukDateInputItem"
    >
      <label
        class="govukLabel lbhLabel"
        for="date-input-year"
      >
        Year
      </label>
      <input
        class="govukInput lbhInput govukDateInputInput govukInputWidth4"
        id="date-input-year"
        name="year"
        role="spinbutton"
        type="number"
        value=""
      />
    </div>
  </div>
</div>
`;
